webgpu.h

```diff
@@ -444,6 +444,17 @@ typedef enum WGPUBufferUsage {
 } WGPUBufferUsage;
 typedef WGPUFlags WGPUBufferUsageFlags;
 
+typedef enum WGPUFeatureName {
```

Comment:
could we consider making this a bitmask?

Reply:
Wouldn't that limit us to just 32 features? I think we could easily end up with more than that, especially if vendors/implementations start experimenting with their own internal extensions.

Reply:
64 maybe, but I still don't think that's enough.

Reply:
Another option is to follow Vulkan here: have a structure with boolean fields for features.

Reply:
Does this mean that like Vulkan we're going to add a chained substructure that contains a single boolean for each feature? Adding more member to the structure would break the ABI

Reply:
If a feature is added, in most cases it would need some API changes as well, so some chained structs are inevitable. Would it make sense to make the feature list an exception from this?

Reply:
What benefit does using the chained structs offer us?

Reply:
It's a mechanism we already have to have. So why not use it, instead of trying to invent a separate mechanism for growing the feature lists?

Reply:
Returning a list via chained structs makes the API unnecessarily difficult to use when the concept of a list is fairly straightforward.

We use extension structs in places where additional members are added to descriptors. This makes sense, because while the JS API can simply add additional members for extensions, it's impossible to do the same thing in the C API without breaking ABI compatibility.

It would be a divergence from the JS API to convert the list of features at `GPUAdapter.features` to a chain of structs when the issue of extending without breaking ABI compatibility does not apply. We can do a direct translation of list->list.